16.0 mig commown self troubleshooting

Honeyxilia opened the pull request.

fcayre committed:
Import the commown_self_troubleshooting module into the 12.0 branch

fcayre committed:
Self troubleshooting controller now create tasks, not issues

They no more exist in v12.

legalsylvain committed:
[FIX] commown_self_troubleshooting : move 1000hz-bootstrap-validator into dedicated lib folder

legalsylvain committed:
[FIX] commown_self_troubleshooting : move bootstrap into dedicated lib folder

legalsylvain committed:
[FIX] commown_self_troubleshooting : move fontawesome into dedicated lib folder

legalsylvain committed:
[FIX] commown_self_troubleshooting : move smartwizard into dedicated lib folder

legalsylvain committed:
[FIX] commown_self_troubleshooting : move jquery into dedicated lib folder

fcayre committed:
Catchup with current troubleshooting pages in v10

fcayre committed:
Catchup with last devs on v10 using unit tests

Note that the tests cover the generation of all self troubleshooting
pages but not the posting of the forms for now.

fcayre committed:
Improve self troubleshooting portal access display

In 12.0, portal right sections are h4 with a hr. Use div with mt-x class to separate sections and subsections.

fcayre committed:
Redirect to the new task URL after creating one with the self troubleshooting tool

fcayre committed:
In py3 tb.format_exc does not take an exception as an argument

loganSIO committed:
Add Serenity option wizard

loganSIO committed:
High priority to Serenity tasks

loganSIO committed:
Completely refactor the way contracts are selected on self troubleshooting pages

We now use a meta data file in each troubleshooting use case directory
so that all use cases are self described and their section can be
automatically generated on the portal user home.

legalsylvain committed:
[META] enable flake8

legalsylvain committed:
[META] enable black

legalsylvain committed:
[META] enable pyupgrade

legalsylvain committed:
[META] enable isort

legalsylvain committed:
[META] enable black (new version)

fcayre committed:
Improve self troubleshooting serenity option wording (recall me > call me)

fcayre committed:
Turn the serenity support task template into html

This is for some reason how the content is now stored.
To be done: do the same with the other sel troubleshooting task templates.

fcayre committed:
Move the portal self troubleshooting links from the right sidebar to the middle

fcayre committed:
Fix unit-test crashing because of previous same-branch commits

loganSIO committed:
Add a Contract Termination for Self Troubleshooting page

legalsylvain committed:
[DOC] commown_self_troubleshooting : try to fix documentation

github-grap-bot committed:
[UPD] README.rst

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.1

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.2

fcayre committed:
[commown_self_troubleshooting] Add forgotten noupdate in project data

This is specially important for project names, which are not translatable.

fcayre committed:
Change contract termination self troubleshooting page title

fcayre committed:
Use the contract-choice common step instead of a specific code

This requires a small wording change in the common step, but nothing too bad.

fcayre committed:
Add some contract attributes on the contract-choice common step options

fcayre committed:
Display the commitment end date of the contract on step 1

fcayre committed:
[test,cosmetic] Use more concise notation

fcayre committed:
Fix closed cursor usage in tests

fcayre committed:
Add contract name in the termination task title

fcayre committed:
Automatically fill-in the storable_product_id and lot_id fields when creating a task in contract termination

fcayre committed:
Display contract termination date on contract choice page

This is useful for all self troubleshooting cases.

fcayre committed:
Use the termination date of chosen contract to toggle following steps

loganSIO committed:
Init generic Issue

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.3

loganSIO committed:
Adding cases to inline radio button

loganSIO committed:
Automatically fill-in the storable _product_id and lot_id fileds when creating task in contract termination

loganSIO committed:
Added a case when a customer encounters a breakage

loganSIO committed:
Refactor helping customers with bugs / incidents

loganSIO committed:
Fixed steps conditioning

loganSIO committed:
Fixing typo + special case to computer delivery on common_steps

loganSIO committed:
Added unit test to generic issue pages

fcayre committed:
Remove useless tags in self troubleshooting pages

fcayre committed:
Add direct link to precise order page in generic issur troubleshooting page

fcayre committed:
Explicit linux course demand in the self troubleshooting ticket

fcayre committed:
Fix double div id in step 5 of generic issue troubleshooting

We add an html variable to the step-ship-module template, so that
self troubleshooting implementations can use it with a specific
html intro.

fcayre committed:
Use a stage reference in the forms instead of an id

This is less hack-prone.
Note there was a bug for the FP2 micro case.

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.4

loganSIO committed:
Fixed unique one line text in the description of the support projet

When the form is sent to the support project, it only appears to have one line instead of having line break everytime the text hit the max width of the description field
+
Replace <pre> HTML tag with <div> in issue_template.xml

loganSIO committed:
Removed tag to ship on template step-confirm-change

loganSIO committed:
Make every self_troubleshooting go to stage_received
To avoid mistakes. Ex: sending a component to a wrong adress

loganSIO committed:
Adding a Linux Logo in step 4

loganSIO committed:
Adding 1 star priority to troubleshooting who need to be shipped
We do need to know when an item need to be quickly shipped. Then the troubleshooting will appear in the top of the Support project.

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.5

loganSIO committed:
Protect against contract usurpation

Check that the posted contract id belongs to the same company as the logged in user.

loganSIO committed:
Add FP4 self troubleshooting form for screen protection

Factor out the form template for both fp3 & fp4 and call it in their dedicated forms

loganSIO committed:
Adding FP4 self troubleshooting issue template

Factor out the issue template for both fp3 & fp4 and call it in their dedicated issues.

loganSIO committed:
Add Fairphone 4 screen protection page in unit test

Unit tests crashed after running it. It was because of I didn't add thoses lines related to Fairphone 4 screenprotection in test res partner.

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.6

fcayre committed:
Fix self troubleshooting form submission systematic crash

There was a name error in the controller.
Having automatic tests for the controller part is now an emergency!

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.7

fcayre committed:
Fix permission issue when checking submitted contract id from self troubleshooting

Users cannot read their contract, so we need to use sudo to check the submitted
contract id.

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.8

loganSIO committed:
Adding HTML tags especially <ul>, <li> to format issue description trough tasks

loganSIO committed:
Adding the rest of issue_template as html

loganSIO committed:
Adding german translation for commown_self_troubleshooting

fcayre committed:
[DE] Improve contract choice options appearance

fcayre committed:
Add lang on html node of self troubleshooting pages

... and use it to translate Next/ Prev page button.

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.9

loganSIO committed:
Small improve on issues template

- Removing a () in data/FP/screen/issue_template.xml
- Extracted both street and split it into two lines
- Formatting more_info text box with <pre> to remove all the text on the same line

fcayre committed:
Take all contracts of current partner's company into account in self troubleshooting

This means display the self troubleshooting links on the portal home page and make all
company's contracts selectable in the dropdown.

fcayre committed:
Display serial number of the devices related to contracts in the dropdown

so that B2B customers know which contract their device is related to.

fcayre committed:
Add user doc to find their device serial number

This is useful to find the contract number related to the device.

fcayre committed:
i18n + FR/DE t10n update

fcayre committed:
Fix bad prev/ next step button DE translation

fcayre committed:
Move python code from template to model

Apart from improving code quality, this fixes a dependency problem:
the qweb template was using the quant_ids field, which is defined in
commown_devices, but commown_self_troubleshooting cannot depend on
commown_devices because it depends on commown_self_troubleshooting...

So we use a method override: commown_devices implements an override of
the contract.contract displayable_description method which uses the
quant_ids field.

fcayre committed:
Change contract dropdown description

Remove useless commitment end date and put devices before dates.

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.10

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.11

loganSIO committed:
Generalize screen FP self troubleshooting into a smartphone one

Also remove unused files

loganSIO committed:
Adding screen protection in self troubleshooting for Crosscall & Shiftphone

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.12

loganSIO committed:
Improvement of the comprehension of the text for the linux pc formation

loganSIO committed:
Reorganization of the manifest by alphabetical order

loganSIO committed:
Adding the possibility to put in attachment in self troubleshooting

loganSIO committed:
Adding german translation for the new generic smartphone template in self troubleshooting

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.13

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.14

fcayre committed:
Refactor contract from sale creation

Move a large part of the code to contract.contract instead of sale.order.line.

loganSIO committed:
Adding css style to <pre> tags to wrap text on issue description

fcayre committed:
Fix the contract field in the "contact a human" section of self troubleshooting

It was still using the old contracts computation mecanism, which does not work anymore.

Only compute contracts once and pass it to the contact a human template from the page template.

fcayre committed:
Make it possible to receive no contract from ST forms

This will be useful for an 'other' like self troubleshooting, with no contract required.

fcayre committed:
Add ST meta-data features: a sequence for page/ category order and no contract domain

No contract domain is useful for the (coming-up) page of a general way
to contact Commown, even when user has no pending contract.

fcayre committed:
Add ST page to open an 'Other' support ticket

... by a user who hasn't necessarily an active contract.

fcayre committed:
Fix ST page template invalid syntax for csrf_token

fcayre committed:
Fix custom module dependencies

commown_self_troubleshooting now depends on commown_devices, not the other way around.
This requires moving the contract termination project creation to a lower level module
as commown_devices requires it.
It also simplifies the implementation of the displayable_description method of the
contract entity type, which can be entirely implemented in the commown_sel_troubleshooting
module instead of partly in commown_devices.

fcayre committed:
Fix typo in fp2 battery self troubleshooting issue template

fcayre committed:
Make ST pages work again after layout change

fcayre committed:
Add headless browser tests for some self troubleshooting pages

FP2-related pages were taken as an example.
Contract termination page tests were also added.

fcayre committed:
Use the portal layout for self troubleshooting pages instead of custom

This makes it possible to use tours to test the module in a headless
browser, which is great for testing the javascript behaviour.

This requires:
- the use of odoo's jquery and bootstrap versions
- the upgrade of smartWizard with its new API
- the removal of 1000hz-bootstrap-validator
- the installation of the base_fontawesome module to get the required
  icons, instead of using a copy in current module.

fcayre committed:
Change old wiki link into a discourse one + wording

fcayre committed:
Copy step 0 chosen contract (if any) into the 'Contact a human' contract field

fcayre committed:
fr t10n update

fcayre committed:
ST h4 are now gray to ease visual distinction with h3

fcayre committed:
Hide navigation buttons on the 'contact a human' ST page

fcayre committed:
Change ST 'Other' titles

Previous titles were supposing other ST pages were available, which is
sometimes not true (e.g. someone with no more contract).

fcayre committed:
Improve Generic ST page

- fix bg-danger usage (old bootstrap version)
- add external link indicators.

fcayre committed:
Improve ST common pages: add external link indicator

fcayre committed:
Add breadcrumbs to ST pages

fcayre committed:
Improve radio buttons and select usage using new bootstrap markup

fcayre committed:
Improve contract termination ST page

- replace radio buttons by a selector
- add an external link indicator
- rename some strange variable names

fcayre committed:
Improve Generic ST page: improve attachment paragraph

fcayre committed:
Improve smartphone screen ST page

- add external link indicators
- fix fontawesome icons
- fix radio buttons using new bootstrap markup

fcayre committed:
Improve FP2 micro ST

- fix typos and reindent paragraphs
- fix radio buttons (new bootstrap version)
- turn a 2-entry drop-down into radio buttons

fcayre committed:
Generic assistance ST page is now second-to-last in the portal

It should be used when the device specific pb is not handled.

fcayre committed:
Improve FP3 display ST: fix radio buttons (new bootstrap version)

fcayre committed:
Put contract termination ST button near the end

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.15

loganSIO committed:
Templating the more_info section using a white_space style for every issue_template.xml

loganSIO committed:
Removing word 'suivantes' from /static/src/self_troubleshooting_tour.js

loganSIO committed:
Factor out run_tour method in tests

loganSIO committed:
Add product informations on contract selector of all ST pages

This will be useful for coming theft and loss page for instance.

loganSIO committed:
Explicit stage_ref in ST forms as hidden inputs
The module commown_self_troubleshooting must now be explicit.

loganSIO committed:
Add theft and loss dedicated ST page

We edit the structure of main controller by adding contractual_issue type and date.
We also implemented unit test for JS and Python.
Finally, we add Declared theft and loss columns in project.xml

loganSIO committed:
Explicit project_ref in ST forms as hidden inputs

The default project_ref was support_project. Now there is no more default value.
Also, we put factor out project_ref inputs in a single one on top of the form.

loganSIO committed:
Fixing FP2 battery test

When this test verifies if a p contains "informations complémentaires" it is case sensitive.
So we removed the "i" then it only checks if the p contains the rest of the string.

loganSIO committed:
Implementing a ST screen protection JS test

In both case : display with screen protection and screen protection only

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.16

Commit:
Add relation between lots and contracts.

The relations relies on the relation between pickings, scraps and
contracts.

github-grap-bot committed:
commown_self_troubleshooting 12.0.1.0.17

fcayre committed:
Fix flake8 detected errors or warnings

fcayre committed:
Add new models for self troubleshooting categories and items

The former is used to group items together in a named section of their
portal in a specific order.

The latter can represent an ST page as before, but also an external link
to a user documentation, like, e.g., Commown's discourse. It will make
it possible to mix links with today's ST pages and order them all at will.

fcayre committed:
Replace ugly metadata.py files by ST items and ST categories in the DB

fcayre committed:
Reimplement partner methods with the new models

The methods should give the exact same results, although some tests had
to be fixed as they were using a bad xml ref which was working by
accident.

fcayre committed:
Add tree and form views for self troubleshooting categories and items

github-grap-bot committed:
[UPD] README.rst

fcayre committed:
Open self troubleshooting external links in a new browser window

fcayre committed:
fr and de t10n update

fcayre committed:
Add a test for self troubleshooting item's name_get method

github-grap-bot committed:
[BOT] post-merge updates

fcayre committed:
Don't crash on self troubleshooting pages when user has no phone or email

This is useless as these fields are mandatory in the self troubleshooting html page,
so the user will add values.

github-grap-bot committed:
[BOT] post-merge updates

loganSIO committed:
Add FP5 screen protection form and Fairphone 5 category

loganSIO committed:
Add Javascript test for FP5 screen protection

By checking the option subscription Commown manipulates modules in the form.

github-grap-bot committed:
[BOT] post-merge updates

loganSIO committed:
Generalize the common "ship module" step

This requires a bit of rewording and the text becomes more general:
the part of device being sent is no longer explicit in the ST page text.
It will be explicit in our confirmation email however.

loganSIO committed:
Implement the generalized "ship module" step in all ST pages

Adjust the ST pages (text, order, ...) accordingly and factorize out
a template dedicated to delivery information display in the issue templates.

loganSIO committed:
Reword "Auto-dépannage" for "Besoins d'assistance"

We reworded the Auto-dépannage section to Besoins d'assistance in the customer portal.
So we do need to reword it also in the ST forms and issue_template.
And Mes demandes d'assistance became Mes interactions avec Commown.

loganSIO committed:
Replace spaces with non-breaking spaces

github-grap-bot committed:
[BOT] post-merge updates

github-grap-bot committed:
Add new commercial request ST page

We received a lot of commercial requests in our support project,
so we decided to create a commercial support project instead.

loganSIO committed:
Add project_ref to the GS DAY ST page form

By adding a JS test for this ST page, then fix the form

github-grap-bot committed:
[BOT] post-merge updates

loganSIO committed:
Add german translation

Also give it a fresh translation for the theft and loss ST page

github-grap-bot committed:
[BOT] post-merge updates

fcayre committed:
Avoid a crash when testing ST with an internal user

... that is already following the project the ST created task is created in.

This is not a supported use case but helps testing, so we add a technical protection
to avoid such crashes.

github-grap-bot committed:
[BOT] post-merge updates

github-grap-bot committed:
[BOT] post-merge updates

fcayre committed:
Rename ST controller method

It was named after the Support project which is no more hard-coded.

fcayre committed:
fix de.po file of the commown_self_troubleshooting module

It was syntactically incorrect, probably due to a wrong automatic rebase.

fcayre committed:
commown_self_troubleshooting de t10n update

github-grap-bot committed:
[BOT] post-merge updates

github-grap-bot committed:
[BOT] post-merge updates

Honeyxilia committed:
Fused 'tool_intro' and 'tool_form' into a standalone template

Honeyxilia committed:
Inserted tool form in Manipulation option step

Honeyxilia committed:
Added specific repair screwdriver for each phone

Honeyxilia committed:
Added manipulation step skip for phones who are not repaired by customers.

Honeyxilia committed:
I18n + de/fr t10n update

Honeyxilia committed:
Included screwdriver_codename variable pass in step-confirm(-module)-change templates + Added new features to other self-troubleshooting forms.

- Added back the custom screwdriver in form to FP2 troubleshoots (camera and microphone)
- Added tool form to FP3 display self-troubleshoot

Honeyxilia committed:
Modified Issue template for Smartphone/screen

Honeyxilia committed:
Added test for whether screwdriver data is set or not

github-grap-bot committed:
[BOT] post-merge updates

Honeyxilia committed:
Added text in templates for screwdriver lending

Honeyxilia committed:
I18n update

github-grap-bot committed:
[BOT] post-merge updates

Honeyxilia committed:
[IMP] : black, isort, prettier

Honeyxilia committed:
[MIG] commown_self_troubleshooting: Migration to 16.0

Honeyxilia committed:
[test] Replaced SavepointCase by TransactionCase

Honeyxilia committed:
[test] Shortened HttpCase import to improve readability

Honeyxilia committed:
Moved asset declaration

Honeyxilia committed:
Inserted the safe_eval wrapped version of datetime module

Honeyxilia committed:
Added frontend assets call at the start of pages

In Odoo 15, the assets functionnality has been revamped at the same time as the web framework change. Among these changes, assets are no longer called in pages declared website.page before they're fully loaded, meaning we can't use JQuery or our functions in JS, and the stylesheets are no longer applied.
So, we call the assets whenever necessary to fully render the self-troubleshooting forms.

Honeyxilia committed:
[test] Fix contract start in contract creation

This properly "starts" the contract - the contract lines are assigned with the _inverse_date_start method from contract.contract, in product_rental.

Honeyxilia committed:
[test] Changed POST response code to a redirect

In the POST request, we pass a redirect argument, meaning we should get redirected after the request succeeds - so, we check that we receive a redirect response code.

Honeyxilia committed:
[test] Replaced depreciated BaseResponse by Response

Honeyxilia committed:
[test] Changed depreciated WSGI application entrypoint

Honeyxilia committed:
[test] Removed test cursor for product.product creation

After the with block was closed, while the product.product record was created, it couldn't be found in database queries (for instance, in the create_contract method, trying to create a contract line with the created product would fail by saying that the product's foreign key doesn't exist in the database table).
However, when creating the product outside of the test cursor, there are no issues in the create_contract method, when creating a contract and its lines.

Honeyxilia committed:
[test] Removed date_end computation after contract creation

In the creation of the contract, if a date_end value is given, then the date_end field of the contract lines are matched with the contract's value. However, using _compute_date_end sets the date_end of both the contract and its lines to False.
So removing it leaves the date_end value we pass it.

Honeyxilia committed:
Adapted to new QWeb template rendering method for new tickets creation

Since the post dict - copied from the request parameters - contains all of the values used in the description - more_info, address info and miscellaneous values (screwdriver...) -, we pass it to the _render method, to be able to retrieve the data.

Honeyxilia committed:
[test] Fixed commercial request test

- The TestPageCommercialRequest did not inherit from any classes, including RunTourMixin, meaning it couldn't use _run_tour - I wonder if this test was being ran in the first place.
=> I made it inherit TestTourWithContractTC to run setUpClass without a contract_name being set, which improves code coverage.
- The tour name was incorrect, and fixing this allows for it to be ran correctly.

Honeyxilia committed:
Changed external xml id retrieval method

Honeyxilia committed:
[test] Renamed Odoo JS module

Honeyxilia committed:
[test] Replaced text actions by click for radio inputs

The text action tries to 'focus' (using focusIn) on the radio input to input the Text - however, it's not possible to use focusIn on radio buttons.
So, we swapped the text actions for inputs for click actions, in order to check the selected action.

Honeyxilia committed:
Fixed contract template domain for Shiftphone contacts

The contract template name for Shiftphones has been changed in production from SHIFT/* to SHIF/*.
While the domain name has been changed in the production of the v12 instance of Odoo, it could be overwritten in the data transfer - so, we make this change to ensure that the domain is already correct.

Honeyxilia committed:
[test] Improve cov.: Added no cover flags over protective code

Honeyxilia committed:
Fixed key navigation disabling

Honeyxilia committed:
Migrated SmartWizard lib to version 6

In Odoo 16, the SmartWizard we use for the self-troubleshooting forms (version 5) started to show some visual glitches. For instance, the progress navbar's transition weren't as smooth (the blocks colors appeared with some delay).
So, we decided to migrate the library to its version 6, released in 2022.

Notes about some changes :
- the state of a step is no longer a unique value, but rather a collection of states that can be activated at once (for instance : a step can both be 'done' (visited) and 'disable', and there are no 'enable' state since not having 'disable' qualifies as being enabled.
So, we have to remove a given state with 'unsetState', and set it with 'setState'

Honeyxilia committed:
Changed and re-implemented Bootstrap classes

fcayre commented on commown_self_troubleshooting/models/troubleshooting.py:
```diff
@@ -1,7 +1,5 @@
-
 from odoo import fields, models
-from odoo.tools.safe_eval import safe_eval
+from odoo.tools.safe_eval import datetime, safe_eval
```
Pourquoi ? Le message de commit ne le dit pas alors que j'imagine qu'il y a une bonne raison. Intuitivement j'aurais plutôt eu tendance à conserver l'import de la bibliothèque standard...

Honeyxilia replied:
Parce que safe_eval requiert des méthodes qui sont attachées à datetime afin de fonctionner en v16.0, et avec la librarie de base, la fonction safe_eval échoue

fcayre commented on commown_self_troubleshooting/controllers/main.py:
```diff
@@ -14,7 +14,14 @@ def ref(self, suffix):
+        return request.env["ir.qweb"]._render(
+            "commown_self_troubleshooting.%s" % post["issue-description-template"],
+            post,
+        )
```
Le message de commit ne me semble pas approprié, il ne décrit pas le changement, qui vise à s'adapter à la nouvelle API de rendu des templates qweb. Tu mets ce contenu dans la docstring et ça ça me paraît très à propos en revanche.

Honeyxilia replied:
J'ai modifié le message c:

fcayre commented on commown_self_troubleshooting/data/Assistance/Generic_Issue/form.xml:
```diff
@@ -9,6 +9,7 @@
 
       <t t-name="website.self-troubleshoot-generic-issue">
         <t t-call="portal.portal_layout">
+          <t t-call-assets="web.assets_frontend" />
```
Le titre du commit contient toujours "TODO".

Honeyxilia replied:
Oui - comme écrit dans la carte Odoo, tu voulais qu'on observe ceci ensemble et du coup j'ai marqué ce commit en TODO afin qu'on regarde ensemble s'il y a une meilleure solution, mais en l'état ça fonctionne même si c'est un peu répétitif

fcayre commented on commown_self_troubleshooting/static/src/js/setup_wizard.esm.js:
```diff
@@ -112,7 +112,9 @@ function setUpWizard($container) {
         selected: 0,
         autoAdjustHeight: false,
-        keyNavigation: false,
+        keyboard: {
```
Le contenu du message de commit est inutile ici, il est plus adapté à un ticket de correction de bug où on discute des choix fonctionnels. Dans les commits, on explique les choix techniques effectués.

Honeyxilia replied:
Fait c:

fcayre commented on commown_self_troubleshooting/tests/test_pages.py:
```diff
@@ -21,17 +21,24 @@ def ts_link_urls(doc):
+    def check_test_prerequisite(cls, check, msg="Prerequisite failed"):
+        "Check the `check` argument is truthy or raise a RuntimeError with give message"
+        if not check:
+            raise RuntimeError(msg)
```
Utiliser setUpClass est utile pour des questions de performance, notamment quand les modifications en base de données sont importantes. Ce n'est pas le cas ici, je pense qu'il vaudrait mieux ne pas le faire et ainsi conserver les appels aux méthodes d'assertion standards.

Honeyxilia replied:
Fait c:

fcayre commented on commown_self_troubleshooting/tests/test_tour.py:
```diff
@@ -26,16 +26,16 @@ def new_open_websocket(self):
 
-class RunTourMixin:
+@odoo.tests.tagged("post_install", "-at_install")
+class RunTourMixin(HttpCase):
```
Ce n'est plus un mixin du coup, il faut renommer la classe, par exemple `RunTourTC`.

Honeyxilia replied:
Fait - j'en ai aussi profité pour renommer la classe TestPageTC, car ça peut amener de la confusion avec PagesTC de test_pages.py

Honeyxilia committed:
[test] Replaced setUp methods by setUpClass

Honeyxilia committed:
[test] Moved HttpCase inheritance

Since RunTourMixin is the class directly calling browser_js, it's better for readability to call it directly in it, and it can help avoid class inheritance oversights.
Since it's not a strict Mixin anymore, we also rename the class

Honeyxilia committed:
[test] Renamed class for better readability

Since this module also contains "pages" related tests (to see what pages are shown in the customer's profile aera), it's better to differentiate this class by denoting that it runs tours.

fcayre commented:
/ocabot merge patch
